### DISCUSSION TITLE
Notifications on Windows Server 2019

### DISCUSSION THREAD

**Discussion by Mohrpheus78** [Windos/BurntToast#152] [General]
I want to use BurntToast on a Windows Server 2019 RDS (Terminalserver) in the user context. It works, but if I delete the start menu folder "Windows Powershell" from the users start menu I don't get notifications anymore. There is no error, it just doesn't work. I only delete the folder with the powershell shortcuts from 
"C:\Users\user\AppData\Roaming\Microsoft\Windows\Start Menu\Programs"

**Comment by Windos** [comment]
You've stumbled onto something a bit deep in the weeds with how toast notifications work.

The notifications API require a shortcut in the Start Menu with an "[Application User Model ID](https://docs.microsoft.com/en-us/windows/win32/shell/appids)" (which I refer to as AppIDs) embedded in them.

This is why creating "custom" sources for toast notifications is [such a hassle](https://toastit.dev/2018/02/04/burnttoast-appid-installer/).

BurntToast defaults to using the AppID for Windows PowerShell (`{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\WindowsPowerShell\v1.0\powershell.exe`) as that tends to be a fairly safe bet.

If the shortcut is removed, the notifications API can't attach the toast to an application and so it fails.

You can attach the toast to another shortcut, the available ones can be found by running `Get-StartApps`, though I _think_ this info comes from the registry and I'm not sure if it updates when the paired shortcut is removed from the Start Menu.

You can supply any of those AppIDs to -AppID on New-BurntToastNotification, or to make it the default, change the AppID in BurntToast's config.json file (remember, updates will overwrite this* and it's a json file so if the AppID has slashes in it they need to be doubled up or otherwise escaped)

_* I need to make it so that you can store a persistent config file outside of the module... will add that to the todo._

**Reply by the-c0d3r** [reply]
Thanks for the answer, it solves my issue as well. I just created a shortcut of powershell and put it at the same startmenu folder, and now my notifications are displayed.

**Reply by anton-hystrix** [reply]
> You've stumbled onto something a bit deep in the weeds with how toast notifications work.
> 
> The notifications API require a shortcut in the Start Menu with an "[Application User Model ID](https://docs.microsoft.com/en-us/windows/win32/shell/appids)" (which I refer to as AppIDs) embedded in them.
> 
> This is why creating "custom" sources for toast notifications is [such a hassle](https://toastit.dev/2018/02/04/burnttoast-appid-installer/).
> 
> BurntToast defaults to using the AppID for Windows PowerShell (`{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\WindowsPowerShell\v1.0\powershell.exe`) as that tends to be a fairly safe bet.
> 
> If the shortcut is removed, the notifications API can't attach the toast to an application and so it fails.
> 
> You can attach the toast to another shortcut, the available ones can be found by running `Get-StartApps`, though I _think_ this info comes from the registry and I'm not sure if it updates when the paired shortcut is removed from the Start Menu.
> 
> You can supply any of those AppIDs to -AppID on New-BurntToastNotification, or to make it the default, change the AppID in BurntToast's config.json file (remember, updates will overwrite this* and it's a json file so if the AppID has slashes in it they need to be doubled up or otherwise escaped)
> 
> _* I need to make it so that you can store a persistent config file outside of the module... will add that to the todo._

This implementation is weird indeed. One problem is when you install PowerShell 7 from WindowsStore you can't use New-BurntToastNotification at all because you can't place the shortcut to it into the start menu normally. So you need to install PowerShell 7 manually or use some kung fu to create shortcut for Store app.

After placing shortcut for Store app nothing happened. It seems I need to reinstall PowerShell from manual installer which seems a lot of work.

At the same time the notifications work for old PowerShell 5 because its shortcut is there in the start menu.

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
